### PR TITLE
Follow GitHub redirects

### LINKED ISSUE #1202: `danger pr` crashes with "undefined method `sha' for nil:NilClass (NoMethodError)"
# Report

## What did you do?

I ran this command, to try and debug locally a Dangerfile behavior on a given PR

```
$ DANGER_GITHUB_API_TOKEN=<mytoken> bundle exec danger pr "https://github.com/MyCompany/my-private-repo/pull/10879" --dangerfile=danger-ci/Dangerfile
```

## What did you expect to happen?

Danger to locally run the rules in the Dangerfile using the metadata of the mentioned PR

## What happened instead?

Crash with following backtrace:

```
Traceback (most recent call last):
  15: from /Users/me/.rvm/gems/ruby-2.6.4/bin/ruby_executable_hooks:24:in `<main>'
  14: from /Users/me/.rvm/gems/ruby-2.6.4/bin/ruby_executable_hooks:24:in `eval'
  13: from /Users/me/.rvm/gems/ruby-2.6.4/bin/danger:23:in `<main>'
  12: from /Users/me/.rvm/gems/ruby-2.6.4/bin/danger:23:in `load'
  11: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/bin/danger:5:in `<top (required)>'
  10: from /Users/me/.rvm/gems/ruby-2.6.4/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
   9: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/commands/pr.rb:60:in `run'
   8: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/commands/pr.rb:60:in `new'
   7: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/danger_core/environment_manager.rb:30:in `initialize'
   6: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/danger_core/environment_manager.rb:30:in `new'
   5: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/ci_source/local_git_repo.rb:45:in `initialize'
   4: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/ci_source/local_git_repo.rb:95:in `found_pull_request'
   3: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/ci_source/support/pull_request_finder.rb:17:in `call'
   2: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/ci_source/support/pull_request_finder.rb:31:in `check_if_any_pull_request!'
   1: from /Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/ci_source/support/pull_request_finder.rb:52:in `pull_request'
/Users/me/.rvm/gems/ruby-2.6.4/gems/danger-6.3.1/lib/danger/ci_source/support/pull_request_finder.rb:83:in `generate_remote_pull_request': undefined method `sha' for nil:NilClass (NoMethodError)
```

The PR url is a copy/paste from the window open in my browser so it should be correct, the Pull request still exists and is still open, and both the base and head branches are still existing on the remote.

`base` branch name is of the form `devname/sdk/feature1` and `head` branch name is on the form `devname/sdk/feature1-part2`

## Your Environment

* Which CI are you running on?

  This was when attempting to run Danger locally, on a PR from our private GitHub repo. Using RVM with Ruby 2.6.4 locally.

* Are you running the latest version of Danger?

  Yes, just updated to 6.3.1

* What is your Dangerfile?

Too long to paste here (it is even split in multiple files that we include using `require_relative`), and I don't think it's relevant to the problem because the crash happens before the `Dangerfile` is even interpreted, at the step where Danger tries to find the Pull Request reference to run its rules on.

### PR DESCRIPTION
**Pull request by telyn**
Fixes #1202

### CODE REVIEW

**Comment by danger-public** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Tests were not updated</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by telyn** [comment]
Tested locally using pry and https://github.com/telyn/test-repo

File: lib/danger/commands/local.rb [hunk 1]
```diff
@@ -66,12 +66,15 @@ def run
 
     private
 
+    # this method is a duplicate of Commands::PR#configure_octokit
+    # - worth a refactor sometime?
```
Comment (AliSoftware) [inline lib/danger/commands/local.rb]: iirc I think Commands::PR was deprecated a while ago in favor of Commands::Local anyway, so I'm assuming `danger pr` should disappear and be removed completely one day even?

**Comment by orta** [comment]
Nice one, yeah, that looks like it should do it 👍

File: lib/danger/commands/local.rb (hunk 1 above)
Comment (telyn) [reply lib/danger/commands/local.rb]: I think it's the other way round - from the danger help:

```
    + local                             Run the Dangerfile locally. This command is
                                        generally deprecated in favor of `danger pr`.
```

**Comment by telyn** [comment]
> Wondering if there's a way to write a unit test for something like that? 🤔
> 
> Thanks for the PR btw 🎉

I think it probably would be, but it'd be a bit gross and rely on the internals of Octokit and on us continuing to use the `Faraday::RackBuilder` thing. It doesn't feel entirely worth it to me (would be happy to be shown a unit test that doesn't seem gross though!

File: lib/danger/commands/local.rb (hunk 1 above)
Comment (AliSoftware) [reply lib/danger/commands/local.rb]: Ah right yep I need some sleep, it's the other way around indeed 👍

still, probably why it's a copy/paste here as local is eventually planned to be removed entirely one day 🙂

**Pull request merged**